### PR TITLE
Use REST connector endpoint for project milestones

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/11092

## Testing
**URL to test:** 
https://deploy-preview-950--atd-moped-main.netlify.app/moped/projects/1912?tab=activity_log

**Steps to test:**

From the timeline tab, add a new milestone to a project. "Added a new milestone: Cabinet set"
Edit a milestone's fields, you should the fields listed out, like:  "Edited a milestone's completion, description"
Delete a milestone. 

you may notice that what is presented here in the activity log doesn't match what John's mockups are in the issue. I've refactored the activity log work to look more like what the issues are requesting. That PR is here: https://github.com/cityofaustin/atd-moped/pull/949

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

**Review by roseeichelmann — APPROVED** [review]
looks good! 🚀🚀🚀

**Review by mddilley — APPROVED** [review]
Looks great - really loving the direction the activity log is going! It is going to make adding new fields and their formatting reeeaaalll nice. 🚢

File: moped-editor/src/utils/activityLogFormatters/mopedMilestonesActivity.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  // delete an existing milestone
+  if (change.description[0].field === "is_deleted") {
+    changeDescription = "Deleted the milestone: ";
+    changeValue = milestoneList[change.record_data.event.data.new.milestone_id];
+
+    return { changeIcon, changeDescription, changeValue };
+  }
+
+  // Multiple fields in the moped_proj_funding table can be updated at once
```
Comment (tillyw) [inline moped-editor/src/utils/activityLogFormatters/mopedMilestonesActivity.js:27]: should this say milestones table?

File: moped-editor/src/utils/activityLogFormatters/mopedMilestonesActivity.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const oldRecord = change.record_data.event.data.old;
+
+  let changes = [];
+
+  // loop through fields to check for differences, push label onto changes Array
+  Object.keys(newRecord).forEach((field) => {
+    if (newRecord[field] !== oldRecord[field]) {
+      changes.push(entryMap.fields[field]?.label);
+    }
+  });
```
Comment (tillyw) [inline moped-editor/src/utils/activityLogFormatters/mopedMilestonesActivity.js:39]: nice!! :sparkles:

File: moped-editor/src/utils/activityLogHelpers.js [hunk 3]
```diff
@@ -1,8 +1,10 @@
 import { formatProjectActivity } from "./activityLogFormatters/mopedProjectActivity";
 import { formatTagsActivity } from "./activityLogFormatters/mopedTagsActivity";
 import { formatFundingActivity } from "./activityLogFormatters/mopedFundingActivity";
+import { formatMilestonesActivity } from "./activityLogFormatters/mopedMilestonesActivity";
 import { formatPartnersActivity } from "./activityLogFormatters/mopedPartnersActivity";
 
+
```
Comment (tillyw) [inline moped-editor/src/utils/activityLogHelpers.js:7]: just a extra line!

**Review by tillyw — APPROVED** [review]
this looks great!! i love how it renders the changes as a list. excited to see this update!

File: moped-editor/src/utils/activityLogFormatters/mopedMilestonesActivity.js (hunk 1 above)
Comment (chiaberry) [reply moped-editor/src/utils/activityLogFormatters/mopedMilestonesActivity.js:27]: eek my impatient copy and pasting habits -- thanks ill fix it

**Pull request merged**